pyanaconda/installation.py

```diff
@@ -234,22 +235,17 @@ def _prepare_installation(payload, ksdata):
     installation_queue.task_completed.connect(lambda x: progress_step(x.name))
 
     # This should be the only thread running, wait for the others to finish if not.
+    #
+    # We need to do this before we start filling the task queue, or else the DBus
+    # tasks we create might have stale data due to background processing threads
+    # still running at task creation time.
     if threadMgr.running > 1:
-        # it could be that the threads finish execution before the task is executed,
-        # but that should not cause any issues
-
-        def wait_for_all_treads():
-            for message in ("Thread %s is running" % n for n in threadMgr.names):
-                log.debug(message)
+        # show a progress message
+        progressQ.send_message(N_("Waiting for %s threads to finish") % (threadMgr.running - 1))
+        for message in ("Thread %s is running" % n for n in threadMgr.names):
+            log.debug(message)
             threadMgr.wait_all()
```

Comment:
It seems to me that waiting for the threads should go out of the for loop  ?

Reply:
It should. Good catch!

Reply:
Oh, right - I somehow thought this is intentional (and this code was the same before) and `wait_all()` returns every time a thread terminates, so you get a nice listing of threads as they terminate. But nope, `wait_all()` does indeed wait for all currently running non-main threads, so it really is pointless to all it like this. :D Lets fix that. :)

Reply:
Looking even more closely, the old method did not have `wait_all()` in the for cycle - double oops. :P

Reply:
Anyway should be fixed now. ;-)